toolchain/check/handle_function.cpp

```diff
@@ -488,8 +489,60 @@ static auto BuildFunctionDecl(Context& context,
   return {function_decl.function_id, decl_id};
 }
 
+CARBON_DIAGNOSTIC(
+    UnusedMarkerInDeclaration, Error,
+    "`unused` marker cannot be used on a declaration that is not a "
+    "definition");
```

Comment:
Could this be inside the function where it's used? Typically we want variables to be declared as close to where they're used as possible, eschewing globals.